avo-hq/avo
# feature: countless pagination

# Description


Fixes #1748 

Added new DSL to resource:
```ruby
self.pagination = -> do
  {
    type: :countless, # default -> :normal
    size: []          # default -> [1, 2, 2, 1]
  }
end
```


```ruby
class CourseLinkResource < Avo::BaseResource
  ...
  self.pagination = -> do
    {
      type: :countless,
      size: []
    }
  end

  # OR

  self.pagination = {
    type: :countless,
    size: []
  }
  ...
end
```


![image](https://github.com/avo-hq/avo/assets/69730720/e2403846-9027-4df8-9889-1f74966b5d56)


![image](https://github.com/avo-hq/avo/assets/69730720/55a9ef3c-a167-4a0c-a6b0-62b9c8fa4c13)

![image](https://github.com/avo-hq/avo/assets/69730720/0dda9eb4-433a-4557-9f4c-d586574e5da6)

![image](https://github.com/avo-hq/avo/assets/69730720/c8b8831c-32f5-48ca-8656-a0adc40bf64b)




# Checklist:


  - https://github.com/avo-hq/avodocs/pull/126 


## Manual review steps


1. Add `self.pagination = -> do
            {
              type: :countless,
              size: []
            }
          end` to any resource
2. Verify resource's index page

Manual reviewer: please leave a comment with output from the test if that's the case.
